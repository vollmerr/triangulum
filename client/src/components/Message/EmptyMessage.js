import React from 'react';
import { Button, Header } from 'semantic-ui-react';

const EmptyMessage = ({ message }) => {
  const centerProps = {
    style: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center',
      textAlign: 'center',
      minHeight: '100vh',
      marginTop: '-10vh',
    },
  };

  const messageProps = {
    as: 'h3',
    textAlign: 'center',
    color: 'violet',
    content: message,
    style: { padding: '20px 0' },
  };

  const buttonProps = {
    icon: 'plus',
    type: 'button',
    color: 'violet',
    content: 'New Crawl',
    onClick: () => { window.location.hash = '/'; },
  };

  return (
    <div {...centerProps}>
      <Header {...messageProps} />
      <Button {...buttonProps} />
    </div>
  );
};

export default EmptyMessage;
