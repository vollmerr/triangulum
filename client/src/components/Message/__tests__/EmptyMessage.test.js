import React from 'react';
import { shallow } from 'enzyme';
import { Button } from 'semantic-ui-react';

import EmptyMessage from '../EmptyMessage';

const props = {
  message: 'test message',
};

let wrapper;
describe('EmptyMessage', () => {
  beforeEach(() => {
    wrapper = shallow(<EmptyMessage {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should redirect on click', () => {
    window.location.hash = '#/wrong';
    wrapper.find(Button).simulate('click');
    expect(window.location.hash).toEqual('#/');
  });
});
