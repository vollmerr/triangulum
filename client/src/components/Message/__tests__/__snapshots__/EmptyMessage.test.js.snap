// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyMessage should render correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "marginTop": "-10vh",
      "minHeight": "100vh",
      "textAlign": "center",
    }
  }
>
  <Header
    as="h3"
    color="violet"
    content="test message"
    style={
      Object {
        "padding": "20px 0",
      }
    }
    textAlign="center"
  />
  <Button
    as="button"
    color="violet"
    content="New Crawl"
    icon="plus"
    onClick={[Function]}
    role="button"
    type="button"
  />
</div>
`;
